### PR TITLE
chore: Remove Eventual Disruption cloud provider interface

### PR DESCRIPTION
**Pull request by engedaam**
Fixes #N/A 

**Description**
- Dropping Eventual Disruption as the use case is not defined well. We will take up an RFC to implement this feature if the customer use case is found.

**How was this change tested?**
- `make presubmit`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/apis/v1/nodepool_budgets_test.go
Comment (jmdeal) [inline pkg/apis/v1/nodepool_budgets_test.go:1]: Looks like we need to still drop the custom reason from the provider on line 81.

**Review by charliedmcb — APPROVED** [review]
Overall, lgtm.

Excited to get the convo started around this area, and ensure we get the best working solution implemented :)

FYI: changes look good, but for `pkg/controllers/disruption/drift_test.go` I only did a quick pass over it.

**Review by jonathan-innis — APPROVED** [review]
/lgtm
/approve

**Pull request merged**